jurigged/codetools.py

```diff
@@ -802,6 +850,51 @@ def reevaluate(self, new_node, glb):
         self._codeobj = new_obj.__code__
         return new_obj
 
+    def reevaluate(self, new_node, glb):
+        # patch: The assertion rewrite is from pytest. Jurigged doesn't
+        #              seem to have a way to add rewrite hooks
+        # TODO: We should fix this in jurigged itself, but porting over from here for now:
+        # https://github.com/breuleux/jurigged/issues/29
+        new_node = self.apply_assertion_rewrite(new_node, glb)
+        obj = self._reevalute(new_node, glb)
+
+        return obj
+
+    def apply_assertion_rewrite(self, ast_func, glb):
+        from _pytest.assertion.rewrite import AssertionRewriter
+
+        # We only want to apply assertion rewrites if we're running in a pytest context
+        if not CONFIG.rewrite_asserts_to_pytest_asserts:
+            return ast_func
```

Comment:
I think this will work as long as its off by default. One consideration is that a more robust solution is the ability to register hooks for reevaluate, then you would loop through those. I would for sure have that in a separate PR.

Reply:
Another approach would be to create a pytest plugin that is automatically registered and ran (which I think is possible) and it triggers this logic to be enabled. That's probably the most robust approach since you would only enable this if you had pytest, and I can't imagine you not wanting it enabled if you did.